openslides/utils/class_register.py

```diff
@@ -0,0 +1,39 @@
+class RegisterSubclasses(type):
+    """
+    Metaclass to track subclasses of a class.
+
+    The metaclass adds an attribute 'subclasses' to the metaclass. This
+    attribute is a set containing all derivated classes.
+
+    After a new subclass is created, the classmethod on_subclass_created is
+    called.
+
+    Example:
+    class Base(metaclass=RegisterSubclasses):
+        pass
+
+    class Child(Base):
+        pass
+
+    assert Child in Base.subclasses
+    """
+
+    def __init__(cls, *args, **kwargs):
+        # Try to add the cls to cls.subclasses. If it does not have the
+        # attribute subclasses, then this is the base class, which should not
+        # be registered
+        try:
+            cls.subclasses.add(cls)
+        except AttributeError:
+            cls.subclasses = set()
+        else:
+            cls.on_subclass_created()
+        super().__init__(*args, **kwargs)
```

Comment:
What do you mean? If you ask if the super call is necessary, then probably not, because I think that type.__init__ does nothing.